IBM Z DFLTCC: Test with MSan

iii-i opened the pull request:
* Add a CI job.
* Do not collect coverage: LLVM's gcov support (part of compiler-rt)
  cannot be built with the MSan instrumentation, which means that
  whenever it's called (in particular, in order to write the results to
  a file at the end), there is a risk of false positives.
* Add __msan_unpoison() calls to DFLTCC inline assembly.
* Make parameter block sizes symbolic constants.
* Move dfltcc() definition after struct dfltcc_param_v0 definition.

iii-i committed:
IBM Z DFLTCC: Test with MSan

* Add a CI job.
* Do not collect coverage: LLVM's gcov support (part of compiler-rt)
  cannot be built with the MSan instrumentation, which means that
  whenever it's called (in particular, in order to write the results to
  a file at the end), there is a risk of false positives.
* Add __msan_unpoison() calls to DFLTCC inline assembly.
* Make parameter block sizes symbolic constants.
* Move dfltcc() definition after struct dfltcc_param_v0 definition.

nmoinvaz commented:
Side-note: You can replace the top 4 includes in `dfltcc_detail.h` with `#include "../../zbuild.h"`

iii-i commented:
> Side-note: You can replace the top 4 includes in `dfltcc_detail.h` with `#include "../../zbuild.h"`

I think `stdio.h` needs to be kept, because `zbuild.h` includes it conditionally, and I need it here for `sprintf()`.

iii-i committed:
IBM Z DFLTCC: Simplify includes in dfltcc_detail.h

Include zbuild.h instead of the standard headers. Keep stdio.h, since
it's provided only conditionally.

Suggested-by: Nathan Moinvaziri <nathan@nathanm.com>

Dead2 reviewed (approved):
LGTM
